pyomo/dae/flatten.py

```diff
@@ -234,47 +239,108 @@ def slice_component_along_sets(
         yield (), c_slice
 
 
-def generate_sliced_components(b, index_stack, slice_, sets, ctype, index_map):
-    """
-    Recursively generate sliced components of a block and its subblocks, along
-    with the sets that were sliced for each component.
-
-    `b` is a _BlockData object.
+def generate_sliced_components(
+    b,
+    index_stack,
+    slice_,
+    sets,
+    ctype,
+    index_map,
+    active=None,
+):
+    """Recursively generate slices of the specified ctype along the
+    specified sets
+    
+    Parameters
+    ----------
+
+    b: _BlockData
+        Block whose components will be sliced
+
+    index_stack: list
+        Sets above ``b`` in the block hierachy, including on its parent
+        component, that have been sliced. This is necessary to return the
+        sets that have been sliced.
+
+    slice_: IndexedComponent_slice or _BlockData
+        Slice generated so far.  This function will yield extensions to
+        this slice at the current level of the block hierarchy.
+
+    sets: ComponentSet of Pyomo sets
+        Sets that will be sliced
+
+    ctype: Subclass of Component
+        Type of components to generate
+
+    index_map: ComponentMap
+        Map from (some of) the specified sets to a "representative index"
+        to use when descending into subblocks. While this map does not need
+        to contain every set in the sliced sets, it must not contain any
+        sets that will not be sliced.
```

Comment:
Missing documentation for `active` (mildly ironic, given the PR title... ;) )

Reply:
I was probably avoiding actually having to write a succinct description of how this flag works. Thanks for adding @blnicho